ledger/participant-integration-api/src/test/lib/scala/platform/store/interning/MockStringInterning.scala

```diff
@@ -64,4 +64,10 @@ class MockStringInterning extends StringInterning {
       override def tryExternalize(id: Int): Option[Party] =
         rawStringInterning.tryExternalize(id).map(Party.assertFromString)
     }
+
+  private[store] def reset(): Unit = synchronized {
```

Comment:
I am wondering why do we need this :thinking: 
The tests passing without.
Do you think we need this?

Thinking more about it: is it maybe the "filter correctly with unknown parties and templates" case? There we already fill this info in MockStringInterning as we persist the dto-s. So the test is not running on code paths, where request params are not able to looked up (internalizing always succeeds). For those code paths to test you would need some more advanced trickery, with rolling an EventStorageBackend with a custom/fixture StringInterning.

Reply:
I have added it after debugging a test and realizing that `StorageBackendProvider.backend.stringInterningSupport` is a stateful object that is shared between tests. I wanted each test to be fully isolated. The change is not necessary to make any test pass.

Reply:
Added a comment in 10e7e7e13af14893b7eed6af8bfefdd9b792c9d8